GeorgiaTech/Api/Controllers/VolumeController.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Server;
+using Api.Models;
+using Server.Controllers;
+// For more information on enabling Web API for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace Api.Controllers
+{
+    [Route("[controller]")]
+    public class VolumeController : Controller
+    {
+        private readonly Server.Controllers.VolumeController volumeController;
+        public VolumeController(GTLContext context)
+        {
+            volumeController = new Server.Controllers.VolumeController(context);
+        }
+        // GET: /volume
+        [HttpGet]
+        public IActionResult Get(int materialId)
+        {
+            Console.WriteLine(materialId);
+            try
+            {
+                List<Server.Models.Volume> volumes; ;
+                if (materialId > 0)
+                {
+                    volumes = volumeController.FindVolumesForMaterial(materialId);
+
+                } else
+                {
+                    volumes = volumeController.FindAll();
+                }
+                var volumeModels = volumes.Select(volume => BuildVolume(volume)).ToList();
+                return Ok(volumeModels);
+
+            }
+            catch (Exception ex)
+            {
+                return NotFound(ex.Message);
+            }
+          
+        }
+
+        public Models.Volume BuildVolume(Server.Models.Volume volume)
+        {
+            var modelVolume = new Volume();
+            modelVolume.VolumeId = volume.VolumeId;
+
+            if (volume.HomeLocation != null)
+            {
+                var homeAddress = new Models.Address
```

Comment:
This should be made through the interface for the Address - to ensure you have dependency inversion

Reply:
This is not the Entity Address model, but the API one.